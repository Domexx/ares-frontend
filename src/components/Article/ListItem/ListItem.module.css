/* === === === === Article/ListItem === === === === */

.ListItem {
  position: relative;
  display: flex;
  flex-direction: column;
  padding: 20px;
}

.ListItem > span {
  z-index: -1;
  border-radius: 12px;
}

.ListItem-image {
  opacity: 0.5;
}

.ListItem:nth-of-type(1) {
  grid-row: 1 / span 2;
  grid-column: 2 / span 4;
}

.ListItem:nth-of-type(2) {
  grid-row: 1 / span 2;
  grid-column: 6 / span 4;
}

.ListItem:nth-of-type(3) {
  grid-row: 1;
  grid-column: 10 / span 4;
}

.ListItem:nth-of-type(4) {
  grid-row: 2;
  grid-column: 10 / span 4;
}

.ListItem-container {
  display: flex;
  flex: 1 1 auto;
  justify-content: space-between;
}

.ListItem-category {
  max-height: 35px;
  padding: 4px 15px 4px 15px;
  border-radius: 20px;
  font-size: 13px;
  background-color: rgb(255 255 255/25%);
}

.ListItem-date {
  font-size: 13px;
}

.ListItem-description {
  font-size: 11px;
}

@media (--laptop) {
  .ListItem:nth-of-type(1) {
    grid-row: 1 / span 3;
    grid-column: 2 / span 6;
  }

  .ListItem:nth-of-type(2) {
    grid-row: 1 / span 3;
    grid-column: 8 / span 6;
  }

  .ListItem:nth-of-type(3) {
    grid-row: 4 / span 3;
    grid-column: 2 / span 6;
  }

  .ListItem:nth-of-type(4) {
    grid-row: 4 / span 3;
    grid-column: 8 / span 6;
  }
}

@media (--tablet) {
  .ListItem {
    grid-column: 1 / span 16 !important;
  }

  .ListItem:nth-of-type(1) {
    grid-row: 1;
  }

  .ListItem:nth-of-type(2) {
    grid-row: 2;
  }

  .ListItem:nth-of-type(3) {
    grid-row: 3;
  }

  .ListItem:nth-of-type(4) {
    grid-row: 4;
  }
}
