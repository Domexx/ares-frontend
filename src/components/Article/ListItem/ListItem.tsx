import React from 'react';
import Image from 'next/image';
import s from './ListItem.module.css';

import { format } from '../../../shared/utils/date';

export type Props = {
  title: string;
  description: string;
  image: string;
  date: number;
  category: string;
};

const ListItem: React.FC<Props> = ({ title, description, image, date, category }) => {
  return (
    <div className={`${s['ListItem']}`}>
      <Image className={s['ListItem-image']} src={image} layout="fill" objectFit="cover" alt={title} />
      <div className={`${s['ListItem-container']}`}>
        <div className={s['ListItem-category']}>{category}</div>
        <div className={s['ListItem-date']}>{format(date, 'EEEE dd LLLL')}</div>
      </div>
      <div className={s['ListItem-title']}>{title}</div>
      <div className={s['ListItem-description']}>{description}</div>
    </div>
  );
};

export default ListItem;
