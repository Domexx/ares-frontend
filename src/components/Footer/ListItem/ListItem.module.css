/* === === === === Footer/ListItem === === === === */

.ListItem {
  margin-top: 24px;
  font-size: 12px;
  text-decoration: none;
  color: var(--clr-text-tertiary);
  transition: opacity 0.5s;
}

.ListItem:hover {
  opacity: 0.7;
}

.ListItem a {
  text-decoration: none;
  color: inherit;
  outline: none;
}
