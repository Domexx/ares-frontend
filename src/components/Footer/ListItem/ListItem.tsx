import React from 'react';
import clsx from 'clsx';
import s from './ListItem.module.css';

import AutoExternalLink from '../../../elements/AutoExternalLink';

export type Props = {
  name: string;
  url: string;
  className?: string;
};

const ListItem: React.FC<Props> = ({ name, url, className }) => {
  const classes = clsx(s['ListItem'], className);

  return (
    <li data-current={name} className={classes}>
      <AutoExternalLink url={url} text={name} />
    </li>
  );
};

export default ListItem;
