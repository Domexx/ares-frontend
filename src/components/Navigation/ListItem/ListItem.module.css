/* === === === === Navigation/ListItem === === === === */

.ListItem {
  display: flex;
  align-items: center;
}

.ListItem {
  overflow: hidden;
  height: 57px;
  padding: 15px 20px;
  border-radius: 5px;
  font-size: 14px;
  font-weight: 500;
  text-decoration: none;
  color: var(--clr-text-tertiary);
  transition: background 250ms ease-in-out;
}

.ListItem:hover {
  background: var(--clr-dark-300);
}

.ListItem-active,
.ListItem-active:hover {
  background: var(--clr-clickable);
  color: var(--clr-dark-500);
}

.ListItem + .ListItem {
  margin-top: 10px;
}

.ListItem svg {
  min-width: 20px;
  min-height: 24px;
}

.ListItem a {
  text-decoration: none;
  color: inherit;
  outline: none;
}

.ListItem svg path {
  fill: var(--clr-text-tertiary);
}

.ListItem-active svg path {
  fill: var(--clr-dark-500);
}
