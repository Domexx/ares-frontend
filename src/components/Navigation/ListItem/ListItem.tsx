import React from 'react';
import s from './ListItem.module.css';

import AutoExternalLink from '../../../elements/AutoExternalLink';

import { ReactComponent as News } from '../../../assets/static/images/Navigation/News.svg';
import { ReactComponent as User } from '../../../assets/static/images/Navigation/User.svg';
import { ReactComponent as Users } from '../../../assets/static/images/Navigation/Users.svg';
import { ReactComponent as Statistics } from '../../../assets/static/images/Navigation/Statistics.svg';

export type Props = {
  name: string;
  url: string;
  icon: string;
};

const ListItem: React.FC<Props> = ({ name, url, icon }) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const icons: any = {
    news: <News />,
    user: <User />,
    users: <Users />,
    statistics: <Statistics />,
  };

  const selectedIcon = icons[icon];

  return (
    <li data-current={name} className={s['ListItem']}>
      {selectedIcon}
      <AutoExternalLink url={url} text={name} />
    </li>
  );
};

export default ListItem;
