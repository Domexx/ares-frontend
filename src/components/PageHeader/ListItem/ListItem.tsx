/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import clsx from 'clsx';
import Link from 'next/link';

import s from './ListItem.module.css';

import { ReactComponent as CogIcon } from '../../../assets/static/images/PageHeader/Cog.svg';
import { ReactComponent as BagIcon } from '../../../assets/static/images/PageHeader/Bag.svg';
import { ReactComponent as MessageIcon } from '../../../assets/static/images/PageHeader/Message.svg';
import { ReactComponent as BellIcon } from '../../../assets/static/images/PageHeader/Bell.svg';
import { ReactComponent as SmileIcon } from '../../../assets/static/images/PageHeader/Smile.svg';

export type Props = {
  name: string;
  url: string;
  icon: string;
  className?: string;
};

const ListItem: React.FC<Props> = ({ name, url, icon, className }) => {
  const classes = clsx(s['ListItem'], className);

  const icons = {
    cog: <CogIcon />,
    bag: <BagIcon />,
    messages: <MessageIcon />,
    bell: <BellIcon />,
    smile: <SmileIcon />,
  };

  const selectedIcon = icons[icon];

  return (
    <li data-current={name} className={classes}>
      <Link passHref href={url}>
        <a>{selectedIcon}</a>
      </Link>
    </li>
  );
};

export default ListItem;
