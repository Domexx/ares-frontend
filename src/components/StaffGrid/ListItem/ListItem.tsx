import React from 'react';
import clsx from 'clsx';
import Image from 'next/image';
import s from './ListItem.module.css';

export type Props = {
  look: string;
  motto: string;
  name: string;
  online: boolean;
  rank: string;
};

const ListItem: React.FC<Props> = ({ look, motto, name, online, rank }) => {
  const statusBadge = clsx(
    s['ListItem-status'],
    online ? s['ListItem-statusBadge-online'] : s['ListItem-statusBadge-offline'],
  );

  return (
    <div className={s['ListItem']}>
      <Image
        className={s['ListItem-look']}
        src={`https://mansionplay.com/imaging/?figure=${look}`}
        alt="Avatar"
        width="130"
        height="200"
      />
      <div className={s['ListItem-nameWrapper']}>
        <span className={s['ListItem-name']}>{name}</span>
        <p className={s['ListItem-motto']}>{motto}</p>
      </div>
      <div className={s['ListItem-rankWrapper']}>
        <div className={s['ListItem-rank']}>{rank}</div>
        <span className={statusBadge} />
      </div>
    </div>
  );
};

export default ListItem;
