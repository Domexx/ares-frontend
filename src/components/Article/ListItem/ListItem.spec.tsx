import React from 'react';

import { render } from '@testing-library/react';

import ListItem from './ListItem';

describe('List Item Component', () => {
  it('should render properly', () => {
    const { container } = render(
      <ListItem
        title="Test Title"
        description="Test Description"
        image="https://images.habbo.com/web_images/habbo-web-articles/lpromo_jan21_gen.png"
        date={0}
        category="Habbo"
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
