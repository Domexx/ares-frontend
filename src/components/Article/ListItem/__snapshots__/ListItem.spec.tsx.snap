// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item Component should render properly 1`] = `
<div
  class="ListItem"
>
  <span
    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
  >
    <img
      alt="Test Title"
      class="ListItem-image"
      data-nimg="fill"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
    />
    <noscript />
  </span>
  <div
    class="ListItem-container"
  >
    <div
      class="ListItem-category"
    >
      Habbo
    </div>
    <div
      class="ListItem-date"
    >
      Donnerstag 01 Januar
    </div>
  </div>
  <div
    class="ListItem-title"
  >
    Test Title
  </div>
  <div
    class="ListItem-description"
  >
    Test Description
  </div>
</div>
`;
